src/cpp/statistics/rtps/messages/RTPSStatisticsMessages.hpp

```diff
@@ -193,16 +195,29 @@ inline void set_statistics_submessage_from_transport(
         statistics_pos += RTPSMESSAGE_SUBMESSAGEHEADER_SIZE;
 
         // Set current timestamp and sequence
-        auto submessage = (StatisticsSubmessageData*)(&send_buffer[statistics_pos]);
+        auto current_pos = &send_buffer[statistics_pos];
         Time_t ts;
         Time_t::now(ts);
 
-        submessage->destination = destination;
-        submessage->ts.seconds = ts.seconds();
-        submessage->ts.fraction = ts.fraction();
-        submessage->seq.sequence = sequence.sequence;
-        submessage->seq.bytes = sequence.bytes;
-        submessage->seq.bytes_high = sequence.bytes_high;
+
+        auto fraction = ts.fraction();
```

Comment:
Is there any reason why this member is treated differently?

Reply:
The fraction member lacks the same reference accessor that seconds does, preventing us from accessing the address of the member directly.

I think it would be a safe addition from an ABI point of view if we were to add it, though.

Reply:
Adding a new non-virtual method to a class is not [ABI breaking](https://www.acodersjourney.com/20-abi-breaking-changes/) (this is the reference I use to consult when in doubt)